src/configure/configurers/provisioningConfigurer.ts

```diff
@@ -0,0 +1,267 @@
+    path: string;
+    absPath: string;
+}
+
+const Layer: string = "ProvisioningConfigurer";
+export class ProvisioningConfigurer implements IProvisioningConfigurer {
+    private provisioningServiceClient: IProvisioningServiceClient;
+    private queuedPipelineUrl: string;
+    private refreshTime: number = 5 * 1000;
+    private localGitRepoHelper: LocalGitRepoHelper;
+    private filesToCommit: DraftFile[] = [];
+
+    constructor(localGitRepoHelper: LocalGitRepoHelper){
+        this.localGitRepoHelper = localGitRepoHelper;
+    }
+
+    public async createProvisioningPipeline(provisioningConfiguration: ProvisioningConfiguration, wizardInputs: WizardInputs): Promise<ProvisioningConfiguration>{
+        try {
+            this.provisioningServiceClient =  await ProvisioningServiceClientFactory.getClient(wizardInputs.githubPATToken, wizardInputs.azureSession.credentials);
+            const OrgAndRepoDetails = wizardInputs.sourceRepository.repositoryId.split('/');
+            return await this.provisioningServiceClient.createProvisioningConfiguration(provisioningConfiguration, OrgAndRepoDetails[0], OrgAndRepoDetails[1]);
+        } catch (error){
+            telemetryHelper.logError(Layer, TracePoints.UnableToCreateProvisioningPipeline, error);
+            throw error;
+        }
+    }
+
+    public async getProvisioningPipeline(jobId: string, githubOrg: string, repository: string, wizardInputs: WizardInputs): Promise<ProvisioningConfiguration>{
+       try {
+        this.provisioningServiceClient =  await ProvisioningServiceClientFactory.getClient(wizardInputs.githubPATToken, wizardInputs.azureSession.credentials);
+        return await this.provisioningServiceClient.getProvisioningConfiguration(jobId, githubOrg, repository);
+       } catch (error){
+            telemetryHelper.logError(Layer, TracePoints.UnabletoGetProvisioningPipeline, error);
+            throw error;
+       }
+    }
+
+    public async checkProvisioningPipeline(jobId: string, githubOrg: string, repository: string, wizardInputs: WizardInputs): Promise<ProvisioningConfiguration> {
+        try {
+            const provisioningServiceResponse = await this.getProvisioningPipeline(jobId, githubOrg, repository, wizardInputs);
+            if ( provisioningServiceResponse.result){
+                if ( provisioningServiceResponse.result.status ===  "Queued" ||  provisioningServiceResponse.result.status == "InProgress") {
+                await sleepForMilliSeconds(this.refreshTime);
+                return await this.checkProvisioningPipeline(jobId, githubOrg, repository, wizardInputs);
+                } else if (provisioningServiceResponse.result.status ===  "Failed") {
+                throw new Error(provisioningServiceResponse.result.message) ;
+                } else {
+                    return provisioningServiceResponse;
+                }
+            }else {
+                throw new Error("Failed to receive queued pipeline provisioning job status");
```

Comment:
we should not fail in case we fail to get a result. Rather continue and proceed to next get.

Reply:
But yes, we will have to keep a max number of such tries so that we do not keep re-trying which just keeps failing.

Reply:
I have a counter which run valid for 5 mins. after that we will throw error